python/ccf/read_ledger.py

```diff
@@ -15,13 +15,17 @@ def indent(n):
 
 
 def stringify_bytes(bs):
-    s = bs.decode()
-    if s.isprintable():
-        return s
-    if len(bs) > 0 and len(bs) <= 8:
-        n = int.from_bytes(bs, byteorder="little")
-        return f"<u{8 * len(bs)}: {n}>"
-    return bs
+    try:
+        s = bs.decode()
+        if s.isprintable():
```

Comment:
I'm not convinced this is a good way to pretty print the contents of arbitrary buffers, it will sometimes display short, unprintable sequences as unsigned integers (signed, with a u prefix!?), and printable integers will be shown as strings. I would prefer if we defaulted to something more regular, even if that means having to rely on schema hints to make things nicer for governance.

Reply:
Yeah, we would need some sort of type hints to make this nicer. We could also just print it in hex.

Reply:
We could embed hints about governance tables, because we have to keep that schema extremely stable anyway, and allow users to supply their own hints as well. Defaulting to hex is reasonable, I agree.